app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,228 @@
+      };
+
+      var createBinding = function(roleName, newSubject, projectName) {
+        RoleBindingsService
+          .create(roleName, newSubject, projectName, requestContext)
+          .then(function() {
+            refreshRoleBindingList();
+            showAlert('rolebindingCreate', 'success', newSubject.name + ' was given the role ' + roleName);
+          }, function(err) {
+            showAlert('rolebindingCreateFail', 'error', newSubject.name + ' was not given the role ' + roleName, "Reason: " + $filter('getErrorDetails')(err));
+          });
+      };
+
+      var updateBinding = function(rb, newSubject, projectName) {
+        RoleBindingsService
+          .addSubject(rb, newSubject, projectName, requestContext)
+          .then(function() {
+            refreshRoleBindingList();
+            showAlert('rolebindingUpdate', 'success', newSubject.name + ' was given the role ' + rb.roleRef.name);
+          }, function(err) {
+            showAlert('rolebindingUpdateFail', 'err', newSubject.name + ' was not given the role ' + rb.roleRef.name, "Reason: " + $filter('getErrorDetails')(err));
+          });
+      };
+
+      var selectedTab = {};
+      if($routeParams.tab) {
+        selectedTab[$routeParams.tab] = true; // ex: tab=Group for Groups, pluralized in the template
+      }
+
+      var subjectKinds = MembershipService.getSubjectKinds();
+
+      angular.extend($scope, {
+        selectedTab: selectedTab,
+        user: user,
+        projectName: $routeParams.project,
+        alerts: {},
+        renderOptions: {
+          collapseEventsSidebar: localStorage.getItem('monitoring.eventsidebar.collapsed') === 'true'
+        },
+        emptyMessage: 'Loading...',
+        subjectKinds: subjectKinds,
+        newBinding: {
+          role: '',
+          kind: $routeParams.tab || _.first(subjectKinds).kind || 'User',
+          name: ''
+        },
+        mode: {
+          edit: false
+        },
+        selectTab: function(selected) {
+          $scope.newBinding.kind = selected;
+        }
+      });
+
+      DataService.list('projects', {}, function(resp) {
+        angular.extend($scope, {
+          projects: _.map(resp.by('metadata.name'), function(project) {
+            return project.metadata.name;
+          })
+        });
+      });
+
+      var createModalScope = function(subjectName, kind, role, currentUserName) {
+        var modalScope = {
+          alerts: {},
+          message: 'Are you sure you want to remove "' + role + '" from the ' + formatKind(kind) + ' "' + subjectName + '"?',
+          okButtonText: 'Remove',
+          okButtonClass: 'btn-danger',
+          cancelButtonText: 'Cancel'
+        };
+        if(_.isEqual(subjectName, currentUserName)) {
+          modalScope.message = 'Are you sure you want to remove "' + role + '" from ' + subjectName + ' (yourself)? ';
+          showAlert('currentUserLastRole', 'error', 'This is your last role in the project!', null, modalScope);
```

Comment:
No exclamation points, we don't like to yell at people :)

Maybe a bit more detail on it, something like  'This is your last role in the project. Removing it may completely remove your ability to see this project.'